File: Assignments/A1/python/A1_test.py
import A1

def test_MaximumValue():
    assert A1.MaximumValue(2, 5, 3, 8, 1, 21, 34, 6, 0) == 34
    
def test_CommonIntegerElements():
    nums1=[1, 54, 125, 23, 644, 31, 5, 61]
    nums2=[125, 23, 2, 3, 1001, 22, 5, 234, 31, 94]
    assert A1.CommonIntegerElements(nums1,nums2) == [5, 23, 31, 125]

def test_CommonStringElements():
    str1=["Hello", "All", "Python", "Ap"]
    str2=["Ap", "Programming", "First", "Language", "Microsoft", "All" , "Python" ]
    assert A1.CommonStringElements(str1,str2) == ["All", "Ap","Python"]

def test_CommonElements():
    nums1=[1, 54, 125, 23, 644, 31, 5, 61]
    nums2=[125, 23, 2, 3, 1001, 22, 5, 234, 31, 94]
    assert A1.CommonElements(nums1,nums2) == [ 125, 23, 31, 5 ]

    str1=["Hello", "All", "Python", "Ap"]
    str2=["Ap", "Programming", "First", "Language", "Microsoft", "All" , "Python" ]
    assert A1.CommonElements(str1,str2) == ["All", "Python","Ap"]


if __name__ == "__main__":
    test_MaximumValue()
    test_CommonIntegerElements()
    test_CommonStringElements()
    test_CommonElements()
    

File: Assignments/A1/cpp/A1_test.cpp
#define CATCH_CONFIG_MAIN
#include"catch.hpp"
#include"A1.cpp"

using namespace A1;

TEST_CASE("MaximumValueTest", "[A1]") {
    REQUIRE(MaximumValue(2, 5, 3, 8, 1, 21, 34, 6, 0) ==  34);
}

TEST_CASE("CommonIntegerElementsTest", "[A1]") {
    int nums1[] = { 1, 54, 125, 23, 644, 31, 5, 61 }; 
    int nums2[] = { 125, 23, 2, 3, 1001, 22, 5, 234, 31, 94 }; 
    vector<int> num=CommonIntegerElements(nums1,nums2,8,10);
    vector<int> exp{5, 23, 31 ,125};
    REQUIRE( num ==  exp ) ;
}

TEST_CASE("CommonStringElementsTest", "[A1]") {
    string str1[] = {"Hello", "All", "Cpp", "Ap"}; 
    string str2[] = { "Ap", "Programming", "First", "Language", "Microsoft", "All", "Cpp" }; 
    vector<string> res=CommonStringElements(str1,str2,4,7);
    vector<string> exp{"All", "Ap", "Cpp"};
    REQUIRE ( res == exp ) ;
}

TEST_CASE("CommonElementsTest", "[A1]") {
    string str1[] = {"Hello", "All", "Cpp", "Ap"}; 
    string str2[] = { "Ap", "Programming", "First", "Language", "Microsoft", "All", "Cpp" }; 
    vector<string> res=CommonElements(str1,str2,4,7);
    vector<string> exp1{"All", "Cpp", "Ap"};
    REQUIRE ( res == exp1 ) ;

    int nums1[] = { 1, 54, 125, 23, 644, 31, 5, 61 }; 
    int nums2[] = { 125, 23, 2, 3, 1001, 22, 5, 234, 31, 94 }; 
    vector<int> num=CommonElements(nums1,nums2,8,10);
    vector<int> exp2{125, 23, 31, 5};
    REQUIRE( num ==  exp2 ) ;
}
